### ISSUE TITLE
All controller pod installation panic with nil pointer dereference on upgraded EKS to version 1.27.

### ISSUE THREAD

**Issue by govindj22** [aws-controllers-k8s/community#1888]
**Describe the bug**
The latest SQS Controller [helm chart](public.ecr.aws/aws-controllers-k8s/sqs-chart:1.0.4) installation is failing with a panic on an upgraded EKS cluster. EKS clusters were upgraded to version 1.27.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14127d0]

goroutine 1 [running]:
k8s.io/client-go/discovery.convertAPIResource(...)
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/aggregated_discovery.go:88
k8s.io/client-go/discovery.convertAPIGroup({{{0x0, 0x0}, {0x0, 0x0}}, {{0xc000165560, 0x15}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/aggregated_discovery.go:69 +0x5f0
k8s.io/client-go/discovery.SplitGroupsAndResources({{{0xc0001640f0, 0x15}, {0xc00013aba0, 0x1b}}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...})
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/aggregated_discovery.go:35 +0x2f8
k8s.io/client-go/discovery.(*DiscoveryClient).downloadAPIs(0x1c21f70?)
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:310 +0x47c
k8s.io/client-go/discovery.(*DiscoveryClient).GroupsAndMaybeResources(0xc0004d1540?)
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:198 +0x5c
k8s.io/client-go/discovery.ServerGroupsAndResources({0x205bf70, 0xc000710630})
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:392 +0x59
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources.func1()
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:356 +0x25
k8s.io/client-go/discovery.withRetries(0x2, 0xc0006daea8)
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:621 +0x72
k8s.io/client-go/discovery.(*DiscoveryClient).ServerGroupsAndResources(0x0?)
	/go/pkg/mod/k8s.io/client-go@v0.26.1/discovery/discovery_client.go:355 +0x3a
k8s.io/client-go/restmapper.GetAPIGroupResources({0x205bf70?, 0xc000710630?})
	/go/pkg/mod/k8s.io/client-go@v0.26.1/restmapper/discovery.go:148 +0x42
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper.func1()
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/client/apiutil/dynamicrestmapper.go:94 +0x25
sigs.k8s.io/controller-runtime/pkg/client/apiutil.(*dynamicRESTMapper).setStaticMapper(...)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/client/apiutil/dynamicrestmapper.go:130
sigs.k8s.io/controller-runtime/pkg/client/apiutil.NewDynamicRESTMapper(0xc00054a6c0?, {0x0, 0x0, 0x24ff0a13d815e301?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/client/apiutil/dynamicrestmapper.go:110 +0x182
sigs.k8s.io/controller-runtime/pkg/cluster.setOptionsDefaults.func1(0xc000660230?)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/cluster/cluster.go:217 +0x25
sigs.k8s.io/controller-runtime/pkg/cluster.New(0xc00047c480, {0xc000541850, 0x1, 0x0?})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/cluster/cluster.go:159 +0x18d
sigs.k8s.io/controller-runtime/pkg/manager.New(_, {0xc000660230, 0x0, 0x0, {{0x20577e8, 0xc00009fe40}, 0x0}, 0x0, {0x0, 0x0}, ...})
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.5/pkg/manager/manager.go:351 +0xf9
main.main()
	/github.com/aws-controllers-k8s/sqs-controller/cmd/controller/main.go:91 +0x455
```

**Steps to reproduce**
Deploy latest ACK-SQS-Controller chart on EKS cluster which are upgraded to version 1.27.

**Expected outcome**
Pod should be up and running after helm installation

**Environment**

* Kubernetes version: 1.27
* Using EKS (yes/no), if so version? : Yes, 1.27
* AWS service targeted (S3, RDS, etc.): SQS

**Comment by a-hilaly** [comment]
Hi @govindj22, thank you for reporting this! looks like this is an issue with all the controllers. We'll patch all of them soon.

**Comment by a-hilaly** [comment]
Root cause is very likely client-go/controller-runtime version is outdated and doesn't work with k8s 1.27

**Comment by a-hilaly** [comment]
@govindj22 We just released [v1.0.5](https://github.com/aws-controllers-k8s/sqs-controller/releases/tag/v1.0.5) of `sqs-controller` - could you please test it and provide feedback in this thread?

**Comment by govindj22** [comment]
Hi @a-hilaly  Thanks for the quick turnaround. We tested deployment with new version(v1.0.5) and it is working fine.
